src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    let newState = { ...currentState };
```

Comment:
The variable name 'newState' could be more descriptive. A name like 'stateCopy' would adhere to better naming conventions as per the checklist.